hardware_interface/CMakeLists.txt

```diff
@@ -1,54 +1,78 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(hardware_interface)
 
-find_package(catkin REQUIRED COMPONENTS roscpp)
+# Load catkin and all dependencies required for this package
+find_package(catkin REQUIRED COMPONENTS
+  roscpp
+)
 
-# Declare catkin package
+# Include header-only Boost libraries
+find_package(Boost REQUIRED)
+
+# Declare a catkin package
 catkin_package(
-  CATKIN_DEPENDS roscpp
-  INCLUDE_DIRS include
-  )
+  INCLUDE_DIRS
+    include
+  CATKIN_DEPENDS
+    roscpp
+  DEPENDS
+    Boost # TODO: Should header-only dependencies be DEPENDS or INCLUDE_DIRS?
```

Comment:
The include paths will be automatically exported by this `DEPENDS Boost`.

Additionally, `INCLUDE_DIRS` is for (from [here](http://docs.ros.org/melodic/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package)):

```
CMAKE_CURRENT_SOURCE_DIR-relative paths to C/C++ includes
```

The include directories of `Boost` do not fall into that category.

Reply:
Thanks for clarifying that, I will remove the TODO.

Reply:
TODO removed in def3541